client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/pattern-assembler-preview.tsx

```diff
@@ -53,12 +53,42 @@ const PatternAssemblerPreview = ( { header, sections = [], footer }: Props ) =>
 
 	useEffect( () => {
 		setScrollToSelector( null );
-	}, [ header, sections ] );
+	}, [ header ] );
 
 	useEffect( () => {
 		setScrollToSelector( footer ? 'footer' : null );
 	}, [ footer ] );
 
+	useEffect( () => {
+		const prevSections = prevSectionsRef.current;
+		const setScrollToSelectorByPosition = ( position: number ) => {
+			const patternPosition = header ? position + 1 : position;
+			setScrollToSelector( `.entry-content > .wp-block-group:nth-child( ${ patternPosition } )` );
+		};
+
+		if ( ! prevSections || ! sections.length ) {
+			prevSectionsRef.current = sections;
+			setScrollToSelector( null );
+			return;
+		}
+
+		if ( prevSections.length === sections.length ) {
+			// Ordered
+			const orderedSectionIndex = sections.findIndex( ( section, index ) => {
```

Comment:
> When moving a pattern up/down the preview is not scrolling properly. This can be improved by passing a prop when the user moves a pattern down or up.

This will find the first different pattern. Thus, if you move the pattern down, you will scroll the next one instead of the selected pattern